File: py/day-2/second_solution.py
def solution(inputArr):
  h=0
  v=0
  angle=0
  for input in inputArr:
    c = input.split(' ')
    d = c[0]
    val = int(c[1])
    if (d == 'forward'):
      h += val
      v += (val * angle)
    elif(d == 'up'):
      angle -= val
    elif(d == 'down'):
      angle += val
  return h*v

File: js/day-2/second-solution.js
module.exports = (arr) => {
  let h = 0; let v = 0
  let angle = 0
  arr.map((i) => {
    const c = i.split(' ')
    const d = c[0]
    const val = +c[1]
    if (d == 'forward') {
      h += val
      v += (val * angle)
    } else if (d == 'up') {
      angle -= val
    } else if (d == 'down') {
      angle += val
    }
  })
  return h * v
}
